### ISSUE TITLE
Don't use ccache unless enabled

### ISSUE THREAD

**Issue by Thaodan** [Frogging-Family/wine-tkg-git#803]
The package tries to use ccache if installed however if the user wants to build with ccache enabled he can set it in `makepkg.conf`
or in the package it self in the options array.

**Comment by Tk-Glitch** [comment]
That feature is kinda buggy. We used to take the options array route and ccache broke randomly 4-5 times over 6 months. The way we're handling it currently has been solid so far. We could however add a toggle, and it was actually a note on my to-dos.

Btw, as I could have answered to `https://github.com/Frogging-Family/wine-tkg-git/issues/802`, the project is a few years old now, and many of the design choices made regarding makepkg handling were to answer issues with it. The project was initially just a PKGBUILD with patches, and it was very respectful of makepkg rules - it was planned to end up on the AUR. Those rules ended up being limiting or buggy for the needs of the devs and powerusers I made it for, so I had to make some choices.
There are quite a few hacks in there just to workaround makepkg bugs and limitations or to force it to fulfill our needs. I'm not saying it's perfect by any means, far from it, but there are reasons behind most of the ugly choices made regarding makepkg 😅

**Comment by Tk-Glitch** [comment]
Worse as in not letting makepkg handle it at all?

Wine's build system will use ccache just fine without makepkg getting in the way.

I did try to report a couple makepkg issues and proposed changes, but my request was closed with a simple "nope". So heh, I'm left with my hacks and workarounds. To be fair, I'm not using makepkg myself for my wine needs.

> I think part of the issue is that this pkgbuild tries to accommodate to user building outside of clean chroot or using aur_helpers

It's part of it, yes, but not only that. It's not just a PKGBUILD anymore. The PKGBUILD itself doesn't do much. The scripts animating it are made to work on virtually any distro, and that's obviously a big factor. There's also the need for plugging proton building in, which adds more complexity. An alternative choice would have been to embrace duplication and enter an even worse maintenance hell, but I simply wouldn't have had enough time for that.
